tutor/config.py

```diff
@@ -111,6 +111,10 @@ def get_user(root: str) -> Config:
         config = get_yaml_file(path)
     upgrade_obsolete(config)
     update_with_env(config)
+
+    for name, value in hooks.Filters.CONFIG_USER.iterate():
+        config[name] = value
```

Comment:
question: If we are fetching the secrets from a secret manager, this will hit the manager every time the config is loaded. Should the consumer of CONFIG_USER be caching results to avoid duplicate/unnecessary calls?

Reply:
Thanks for the feedback! In my mind it was going to be on the code that's fetching secrets to implement caching, and then it will return a full list of key:value tuples to the filter itself so as to save complexity on tutor's side (and give users more flexibility in how they actually implement this filter). Thoughts?

Reply:
Yes, makes sense. When I say consumer, I also meant the code that would be using the filter.

Reply:
Ah then yes, I would say the code that is feeding the filter should cache results but ultimately that will be down to the specific implementations of devs and their org's needs/requirements

For the example on the root of the PR `s.do_config()` is reaching out to a singular AWS Secrets Manager secret with many keys in the JSON payload, so its just a single request for us at the end of the day, but other orgs could cache timestamps for the values and compare those as needed prior to making the request!